sklearn/metrics/_scorer.py

```diff
@@ -101,12 +101,23 @@ def __call__(self, estimator, *args, **kwargs):
         cache = {} if self._use_cache(estimator) else None
         cached_call = partial(_cached_call, cache)
 
+        all_scorer_failed = True
         for name, scorer in self._scorers.items():
-            if isinstance(scorer, _BaseScorer):
-                score = scorer._score(cached_call, estimator, *args, **kwargs)
+            try:
+                if isinstance(scorer, _BaseScorer):
+                    score = scorer._score(cached_call, estimator, *args, **kwargs)
+                else:
+                    score = scorer(estimator, *args, **kwargs)
+                scores[name] = score
+            except Exception as e:
+                scores[name] = e
+                last_exception = e
             else:
-                score = scorer(estimator, *args, **kwargs)
-            scores[name] = score
+                all_scorer_failed = False
+
+        if all_scorer_failed:
+            raise last_exception
```

Comment:
Since we are already going through the dictionary in `_score` I prefer to handle the exceptions there. This way `_MultimetricScorer` will only need to set the exceptions in `scores[name]`:

```python
for name, scorer in self._scorers.items():
    try:
        ...
    except Exception as e:
        scores[name] = e

return scores
```

And then in `_score`:

```python
# check errors in `_MultimetricScorer`
if isinstance(scorer, _MultimetricScorer):
    exceptions = {name: isinstance(e, Exception) for name: e in scores.items()}
    if exceptions:
        if error_score == "raise":
            # raise error
        else:
            warnings.warn(f"Scoring failed for...", UserWarning)
            # replace exceptions with error_score
            new_scores = {name: error_score for name in scores}
            scores.update(new_scores)
```

Reply:
Thank you for the review! Pushed the changes.